lib/nine_box.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:ui';
+
+import 'package:flame/palette.dart';
+import 'package:flame/sprite.dart';
+
+/// This allows you to create a rectangle textured with a 9-sliced image.
+///
+/// How it works is that you have a template image in a 3x3 grid, made up of 9 tiles,
+/// and a new rectangle can be draw by keeping the 4 corners, expanding the 4 sides only
+/// in the direction in which they are located and expanding the center in both directions.
+/// That allows you to have non distored borders.
+class NineBox {
+  /// The sprite used to render the box, must be a 3x3 grid of square tiles.
+  Sprite sprite;
+
+  /// The size of each tile in the source sprite image.
+  int tileSize;
+
+  /// The size each tile becomes when rendered (optionally used to scale the src image).
+  int destTileSize;
+
+  /// Creates a nine-box instance.
+  ///
+  /// [srite] is the 3x3 grid and [tileSize] is the size of each tile.
+  /// The src sprite must a square of size 3*[tileSize].
+  ///
+  /// If [tileSize] is not provided, the width of the sprite is assumed as the size.
+  /// Otherwise the width and height properties of the sprite are ignored.
+  ///
+  /// If [destTileSize] is not provided, the evaluated [tileSize] is used instead
+  /// (so no scaling happens).
+  NineBox(this.sprite, {int tileSize, int destTileSize}) {
```

Comment:
I am not sure about the name, NineBox seems a little unclear on what this is, maybe something like NineTilePanel, or something like this?

Also we should provide a component for this as well, to make it easy to use on basegames

Reply:
I thought about adding a component, but I then realized you would probably never have a component with just this? like I use it in other components as composable piece

Reply:
about the name, I can change to NineTilePanel

I think no name is totally clear but yours seems better

Reply:
I can see come good usages for a component like this, for example, if you override this component to draw some text inside it, add the Tapable mixin, you got yourself an easy way to make buttons inside Flame.

Reply:
ok, I will add one